### PR TITLE
Remove unnecessary scopes

### LINKED ISSUE #898: Remove scopes that only exist for internal agent log correlation
There are many places that create scopes who's purpose is unclear, e.g.

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/227bde87ce7aa8f9ae725f2daaf7b0c312117b7e/instrumentation/rabbitmq-2.7/src/main/java/io/opentelemetry/auto/instrumentation/rabbitmq/amqp/RabbitChannelInstrumentation.java#L270-L278

The original purpose of this was so that internal logging within the agent that occurred within this block of code could include the active span info, but that purpose is very hidden and not clear to anyone reading the code.

I suggest we remove them.

An alternative, if we want to keep them, would be to apply this pattern everywhere, and use a wrapper named something like `startLoggingScope` to make it's purpose clear.

### PR DESCRIPTION
**Pull request by trask**
Closes #898

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

File: instrumentation/servlet/servlet-common/src/main/java/io/opentelemetry/instrumentation/auto/servlet/dispatcher/RequestDispatcherInstrumentation.java [hunk 1]
```diff
@@ -115,18 +114,16 @@ public static void start(
         parent = servletContext;
       }
 
-      try (Scope ignored = ContextUtils.withScopedContext(parent)) {
-        span = TRACER.startSpan(method);
+      span = TRACER.startSpan(method);
```
Comment (iNikem) [inline instrumentation/servlet/servlet-common/src/main/java/io/opentelemetry/instrumentation/auto/servlet/dispatcher/RequestDispatcherInstrumentation.java:117]: Here the correct parent span is important. I think this change breaks it.

File: instrumentation/spymemcached-2.12/src/main/java/io/opentelemetry/instrumentation/auto/spymemcached/CompletionListener.java [hunk 2]
```diff
@@ -28,29 +26,27 @@ public CompletionListener(MemcachedConnection connection, String methodName) {
   }
 
   protected void closeAsyncSpan(T future) {
-    try (Scope ignored = currentContextWith(span)) {
-      try {
-        processResult(span, future);
-      } catch (CancellationException e) {
+    try {
+      processResult(span, future);
```
Comment (iNikem) [inline instrumentation/spymemcached-2.12/src/main/java/io/opentelemetry/instrumentation/auto/spymemcached/CompletionListener.java:30]: What if `processResult` creates new spans? Will their parent change now?

**Review by iNikem — CHANGES_REQUESTED** [review]

File: instrumentation/servlet/servlet-common/src/main/java/io/opentelemetry/instrumentation/auto/servlet/dispatcher/RequestDispatcherInstrumentation.java (hunk 1 above)
Comment (trask) [reply instrumentation/servlet/servlet-common/src/main/java/io/opentelemetry/instrumentation/auto/servlet/dispatcher/RequestDispatcherInstrumentation.java]: Eeek, thanks for catching this!

File: instrumentation/spymemcached-2.12/src/main/java/io/opentelemetry/instrumentation/auto/spymemcached/CompletionListener.java (hunk 2 above)
Comment (trask) [reply instrumentation/spymemcached-2.12/src/main/java/io/opentelemetry/instrumentation/auto/spymemcached/CompletionListener.java:30]: `processResult` is only used to populate span attributes, it doesn't make callbacks into the user code, so I think this is ok

**Review by iNikem — APPROVED** [review]

File: instrumentation/servlet/servlet-common/src/main/java/io/opentelemetry/instrumentation/auto/servlet/dispatcher/RequestDispatcherInstrumentation.java (hunk 1 above)
Comment (anuraaga) [reply instrumentation/servlet/servlet-common/src/main/java/io/opentelemetry/instrumentation/auto/servlet/dispatcher/RequestDispatcherInstrumentation.java]: Can we add a test for it?

**Comment by iNikem** [comment]
@trask it seems you have to run spotless on this.

File: instrumentation/servlet/servlet-common/src/main/java/io/opentelemetry/instrumentation/auto/servlet/dispatcher/RequestDispatcherInstrumentation.java (hunk 1 above)
Comment (trask) [reply instrumentation/servlet/servlet-common/src/main/java/io/opentelemetry/instrumentation/auto/servlet/dispatcher/RequestDispatcherInstrumentation.java]: done

**Pull request merged**